docs/src/main/asciidoc/deploying-to-openshift-docker-howto.adoc

```diff
@@ -0,0 +1,154 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+[id="deploying-to-openshift-docker-howto"]
+= Deploy {project-name} Java applications to {openshift} by using a Docker build strategy
+include::_attributes.adoc[]
+:diataxis-type: howto
+:categories: cloud, native
+:summary: This guide describes how to build and deploy a Quarkus application on {openshift} by using the Docker build strategy.
+:topics: devops,kubernetes,openshift,cloud,deployment
+:extensions: io.quarkus:quarkus-openshift
+
+As an application developer, you can deploy your applications to {openshift} by using the Docker build strategy as a deployment option.
+This functionality is provided by the `quarkus-openshift` extension.
+
+The container gets built inside the {openshift} cluster and is provided as an image stream.
+
+Your Quarkus project includes pregenerated Dockerfiles with instructions.
+When you want to use a custom Dockerfile, you must add the file in the `src/main/docker` directory or anywhere inside the module. Additionally, you must set the path to your Dockerfile by using the `quarkus.openshift.jvm-dockerfile` property.
+
+== Prerequisites
+
+* You have OpenJDK {JDK-ver-all} installed.
+* You have set the `JAVA_HOME` environment variable to the location of the Java SDK.
+* You have Apache Maven {maven-version} installed.
+* You have a Quarkus project that includes the `quarkus-openshift` extension.
+* You have access to a {openshift} cluster and the latest compatible version of the `oc` tool installed.
+* You are working in the correct OpenShift project namespace.
+
+== Procedure
+
+. Set the Docker build strategy in your `application.properties` configuration file:
++
+[source, properties]
+----
+quarkus.openshift.build-strategy=docker
+----
+. Optional: Set the following properties in the `application.properties` file, as required by your environment:
+.. If you are using an untrusted certificate, configure the `KubernetesClient`:
++
+[source,properties]
+----
+quarkus.kubernetes-client.trust-certs=true
+----
+.. Expose the service to create an {openshift} route:
++
+[source,properties]
+----
+quarkus.openshift.route.expose=true
+----
+.. Set the path to your custom Dockerfile:
++
+[source,properties,subs="attributes+,+quotes"]
+----
+quarkus.openshift.jvm-dockerfile=<path_to_your_dockerfile>
+----
+The following example shows the path to the `Dockerfile.custom-jvm`:
++
+[source,properties]
+----
+quarkus.openshift.jvm-dockerfile=src/main/resources/Dockerfile.custom-jvm
+----
+
+. Package and deploy your application to the current OpenShift project:
++
+[source,shell,subs="attributes+,+quotes"]
+----
+./mvnw clean package -Dquarkus.openshift.deploy=true
+----
+
+== Verification
+
+The verification steps and related terminal outputs are demonstrated on the `openshift-helloworld` example application.
```

Comment:
```suggestion
The following verification steps use the `openshift-helloworld` example application.
```
Where does this example live? I don't see it in the https://github.com/quarkusio/quarkus-quickstarts...

Reply:
Updated per your suggestion and the example referenced relate to the sentences that follow in the Verfication section.